src/command.rs

```diff
@@ -1,18 +1,52 @@
-use crate::{conv, handle_device_error, make_slice, native, OwnedLabel, GLOBAL};
+use crate::native::{IntoHandleWithContext, UnwrapId, IntoHandle};
+use crate::{conv, handle_device_error, make_slice, native, Context, OwnedLabel};
 use std::ffi::CStr;
 use std::os::raw::c_char;
+use std::sync::Arc;
 use std::{borrow::Cow, num::NonZeroU64};
+use wgc::command::{ComputePass, RenderBundleEncoder, RenderPass};
 use wgc::{
     command::{bundle_ffi, compute_ffi, render_ffi},
     gfx_select,
 };
 
+fn unwrap_render_bundle_encoder<'a>(
+    handle: *mut native::WGPURenderBundleEncoderImpl,
+) -> (&'a mut RenderBundleEncoder, &'a Arc<Context>) {
+    unsafe {
+        let v = handle.as_mut().expect("invalid render bundle encoder");
+        (&mut v.encoder, &v.context)
+    }
+}
+
+fn unwrap_compute_pass_encoder<'a>(
+    handle: *mut native::WGPUComputePassEncoderImpl,
+) -> (&'a mut ComputePass, &'a Arc<Context>) {
+    unsafe {
+        let v = handle.as_mut().expect("invalid compute pass encoder");
+        (&mut v.encoder, &v.context)
+    }
+}
+
+fn unwrap_render_pass_encoder<'a>(
+    handle: *mut native::WGPURenderPassEncoderImpl,
+) -> (&'a mut RenderPass, &'a Arc<Context>) {
+    unsafe {
+        let v = handle.as_mut().expect("invalid render pass encoder");
+        (&mut v.encoder, &v.context)
+    }
+}
+
 #[no_mangle]
 pub unsafe extern "C" fn wgpuCommandEncoderFinish(
     command_encoder: native::WGPUCommandEncoder,
     descriptor: Option<&native::WGPUCommandBufferDescriptor>,
 ) -> native::WGPUCommandBuffer {
-    let command_encoder = command_encoder.expect("invalid command encoder");
+    let (command_encoder, context) = {
+        command_encoder.as_ref().expect("invalid command encoder");
+        let b = Box::from_raw(command_encoder);
```

Comment:
this will drop the `command_encoder` here, was it intended?

Reply:
I think I was slightly confused by this: https://github.com/gfx-rs/wgpu-native/issues/154 
In my code I was calling wgpuCommandEncoderFinish and pass the result to wgpuQueueSubmit and neither the encoder or command buffer had to be dropped. (now they would need to be since they return a box instead of just and id)

I'll change it so that you should explicitly call wgpuCommandEncoderDrop/wgpuCommandBufferDrop

Reply:
> should explicitly call wgpuCommandEncoderDrop

in that case we should make `wgpuRenderBundleEncoderFinish` behave in same way, it will make drop-ing refs consistent.

Reply:
I'm going of the example code and the linked, which seems to expect that command buffers/encoders are either used or dropped (not both)
So I unified the usage so that it works as follows, the following functions drop the corresponding objects:
```
WGPUComputePassEncoder:
- wgpuComputePassEncoderEnd
- wgpuComputePassEncoderDrop

WGPURenderPassEncoder:
- wgpuRenderPassEncoderEnd
- wgpuRenderPassEncoderDrop

WGPURenderBundleEncoder:
- wgpuRenderBundleEncoderFinish
- wgpuRenderBundleEncoderDrop

WGPURenderBundle:
- wgpuRenderBundleDrop

WGPUCommandEncoder:
- wgpuCommandEncoderFinish
- wgpuCommandEncoderDrop

WGPUCommandBuffer:
- wgpuQueueSubmit
- wgpuCommandBufferDrop
```